### ISSUE TITLE
[PhpUnitBridge] weak vendors mode silence legitimate deprecations

### ISSUE THREAD

**Issue by theofidry** [symfony/symfony#21991]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.3.0 (master)

Not sure if that's the expected behaviour, but say if you have a test requesting a private service, you should get a warning along the lines of:

```
Requesting the "PrivateFoo" private service is deprecated since Symfony 3.2 and won't be supported anymore in Symfony 4.0
```

However if you use PhpUnitBridge with `weak_vendors` mode introduced in https://github.com/symfony/symfony/pull/21539, then this deprecation notice will be silenced.

I'm not sure if that's the expected behaviour, my understanding of `weak_vendors` was that third-party deprecations, as a third-party bundle using a deprecated feature, would be silenced, not any deprecation thrown by Symfony itself triggered or not by your code.

<hr />

Steps to reproduce:

- clone https://github.com/theofidry/phpunitbridge-demo2
- `composer install`
- `./tests.sh`

Inside `tests.sh` we can see that with `weak_vendors`, all deprecation notices are caught whereas without it, the deprecation notice for requesting private services is thrown (and is triggered by the user code).

/cc @greg0ire

**Comment by greg0ire** [comment]
> I'm not sure if that's the expected behaviour, my understanding of weak_vendors was that third-party deprecations, as a third-party bundle using a deprecated feature, would be silenced, not any deprecation thrown by Symfony itself triggered or not by your code.

You may close this, it's the expected behavior. Only deprecations that come that a `trigger_error` call that can be found under `/src` will not be silenced. These are deprecations that you wrote yourself. Can you read the documentation and suggest changes to make it clearer maybe?

**Comment by greg0ire** [comment]
To clarify the intended usage is for libraries that want a build job to shield them from in CI errors where they trigger their own deprecations.

**Comment by greg0ire** [comment]
TL;DR: Symfony is a vendor too

**Comment by theofidry** [comment]
Meh, that seems to be a missing mode then but I guess it's hard to have one that would do that 🙄

**Comment by greg0ire** [comment]
@theofidry I thought about it but regexes as a value for `SYMFONY_DEPRECATIONS_HELPER` are already taken for something else.
